.github/workflows/_get-changed-files.yml

```diff
@@ -0,0 +1,48 @@
+name: Get Changed Files
+
+on:
+  workflow_call:
+    outputs:
+      changed-files:
+        description: "List of changed files (space-separated) or '*' if not in a PR"
+        value: ${{ jobs.get-changed-files.outputs.changed-files }}
+
+jobs:
+  get-changed-files:
+    runs-on: ubuntu-latest
+    outputs:
+      changed-files: ${{ steps.get-files.outputs.changed-files }}
+    
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      
+      - name: Get changed files
+        id: get-files
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          # Check if we're in a pull request context
+          if [ "${{ github.event_name }}" = "pull_request" ] || [ "${{ github.event_name }}" = "pull_request_target" ]; then
+            echo "Running in PR context"
+            
+            # Get the PR number from the github context
+            PR_NUMBER="${{ github.event.number }}"
+            
+            # Use gh CLI to get changed files in the PR
+            CHANGED_FILES=$(gh pr view $PR_NUMBER --json files --jq '.files[].path' | tr '\n' ' ' | sed 's/ $//')
```

Comment:
Alternatively, I have seen this logic https://github.com/pytorch/pytorch/blob/306dd19216b656467143483395ef582feb5d7d07/scripts/lint_xrefs.sh#L33 to use git to get the list of changed files on PR by comparing the PR base and head SHAs https://github.com/pytorch/pytorch/blob/306dd19216b656467143483395ef582feb5d7d07/.github/workflows/_link_check.yml#L50

Saving a query to GitHub, probably

Reply:
Oh actually the reason I'm not using an approach like this is because I don't actually want to check out the repository. I removed the `actions/checkout` step that is here since it takes 1min to complete.

Reply:
But this is still a good approach that we should consider if we suddenly run into a ton of rate limits but since GH_TOKEN is unique each time I don't really anticipate that we should run into issues.